src/ui/public/notify/notifier.js

```diff
@@ -111,6 +136,18 @@ function add(notif, cb) {
     return notif.timerId ? true : false;
   };
 
+  // decorate the notification with helper functions for the template
+  notif.getButtonClass = () => typeToButtonClassMap[notif.type];
+  notif.getAlertClassStack = () => `toast-stack alert ${typeToAlertClassMap[notif.type]}`;
+  notif.getIconClass = () => (notif.type === 'banner') ?  '' : `fa fa-${notif.icon}`;
```

Comment:
@cjcenizal the `notif.icon` could be mappable, but the complication is that icon is overridable when the notification is instantiated. Which means there could be a plugin instantiating a notification like:

```
notifier.add({
  type: 'warning',
  content: 'Notification message string for the user.',
  icon: 'beer',  // custom icon
  title: 'Beer Warning Notification',
  actions: ...
}, function callback() { ... });
```

If we refactor the icon to use a mapping the say the button class and alert class are doing, it could potentially be an API breaking change.

I think there is a way to preserve backwards compatibility and have the cleanup that we want (where the goal is to make the icon classes greppable through the code by leaving them as full strings instead of string concatenations), but that could be a bit wider of a refactoring effort. Changing the way that default icons are attributed to each type of notification would be a start.